wright/src/grammar/ast.rs

```diff
@@ -70,6 +70,16 @@ pub struct Identifier<'s> {
     pub frag: Fragment<'s>,
 }
 
+/// A scoped, or qualified, name.
+#[derive(Clone, Debug)]
+pub struct ScopedName<'s> {
+    /// The source code fragment.
+    pub frag: Fragment<'s>,
+    /// The sequence of simple identifiers.
+    /// Example: foo::bar::baz -> [ foo, bar, baz ]
+    pub names: Vec<Identifier<'s>>,
```

Comment:
Since `names.len()` must be >= 1, it might make sense to use another field for the bottom-level `baz` identifier?

```rust
/// foo::bar::baz -> baz
pub last: Identifier<'s>,
/// foo::bar::baz -> [foo, bar, baz]
pub path: Identifier<Identifier<'s>>,
```

Reply:
Maybe. It would emphasize the "namespace + name" aspect

Reply:
Yeah I agree with @io12 here